internal/core/items.rs

```diff
@@ -754,6 +754,25 @@ impl Item for FocusScope {
     }
 }
 
+impl FocusScope {
+    pub fn focus(self: Pin<&Self>, window_adapter: &Rc<dyn WindowAdapter>, self_rc: &ItemRc) {
```

Comment:
I think it's a bit unfortunate that we have to add this function to all builtin items that take focus and then "dispatch" it with the interpreter/ffi.
In a sense, a single  BuiltinFunction::SetFocusItem was not too bad.
Now, since there are only two items that take the focus, this is not too bad.

Reply:
Yeah, it's not ideal. It'll get slightly worse with the Qt style...